ksql-engine/src/main/java/io/confluent/ksql/function/udf/json/JsonExtractStringKudf.java

```diff
@@ -16,62 +16,74 @@
 
 package io.confluent.ksql.function.udf.json;
 
+import com.google.common.collect.ImmutableList;
+
 import com.fasterxml.jackson.databind.JsonNode;
 import com.fasterxml.jackson.databind.ObjectMapper;
-import com.google.common.collect.ImmutableList;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+
+import java.io.IOException;
+import java.util.List;
+
 import io.confluent.ksql.function.KsqlFunctionException;
 import io.confluent.ksql.function.udf.Kudf;
-import io.confluent.ksql.util.KsqlException;
 import io.confluent.ksql.util.json.JsonPathTokenizer;
 
-import java.io.IOException;
-
 public class JsonExtractStringKudf implements Kudf {
   private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
 
-  private String path = null;
-  private ImmutableList<String> tokens = null;
-
-  @Override
-  public void init() {
-  }
+  private List<String> tokens = null;
```

Comment:
as above with the timestamp udfs. `ensureInitialized` can be called by n threads and `tokens != null` could be `false` for all of them

Reply:
As above, don't see this as an issue.